src/connection.js

```diff
@@ -718,8 +722,16 @@ class Connection extends EventEmitter {
     isolationLevel || (isolationLevel = this.config.options.isolationLevel);
     const transaction = new Transaction(name || '', isolationLevel);
     if (this.config.options.tdsVersion < '7_2') {
-      return this.execSqlBatch(new Request('SET TRANSACTION ISOLATION LEVEL ' + (transaction.isolationLevelToTSQL()) + ';BEGIN TRAN ' + transaction.name, callback));
+      const self = this;
+      return this.execSqlBatch(new Request('SET TRANSACTION ISOLATION LEVEL ' + (transaction.isolationLevelToTSQL()) + ';BEGIN TRAN ' + transaction.name, function() {
+        self.transactionDepth++;
+        if (self.transactionDepth === 1) {
+          self.inTransaction = true;
+        }
+        return callback.apply(this, arguments);
```

Comment:
I think my change actually preserves the prior behavior in terms of how 'callback' is being invoked.

In the old code some code is calling 'callback' from somewhere and either giving it a 'this' value or not. We don't know that from here.

In the new code, new inline function()  replaces 'callback' and hence should be called with the same 'this' value as above.  All I'm doing is passing on that 'this' value to the invocation of 'callback', which preserves the old behavior.

Makes sense or am I missing something?